pipeline/05-finalize.R

```diff
@@ -450,12 +474,26 @@ if (params$toggle$upload_to_s3) {
       .[!grepl("=", .)] %>%
       paste0(collapse = "\n")
 
+    # Get a link to the uploaded Quarto report
+    report_path_parts <- strsplit(paths$output$report$s3[1], "/")[[1]]
+    report_bucket <- report_path_parts[3]
+    report_path <- report_path_parts[4:length(report_path_parts)] %>%
+      paste(collapse = "/")
+    # Use direct link to the console instead of to the object so that we don't
+    # have to bother with signed URLs
+    report_url <- paste0(
+      "https://s3.console.aws.amazon.com/s3/object/",
+      "{report_bucket}/{report_path}?region=us-east-1&tab=overview"
+    ) %>%
+      glue::glue()
+
     # Publish to SNS
     pipeline_sns$publish(
       Subject = paste("Model Run Complete:", run_id),
       Message = paste0(
         "Model run: ", run_id, " complete\n",
         "Finished in: ", pipeline_sns_total_time, "\n\n",
+        "Report link: ", report_url, "\n\n",
```

Comment:
In the process of making this change, I realized that I haven't been getting any of these SNS emails recently. Have you? If not, I'll file an issue to investigate why we're missing them; otherwise it's probably just an issue with my subscription or my Outlook config.

Reply:
No I haven't! Good catch, and very weird. There could be something in the params file preventing notification. Or this step could just be misconfigured. Either way, let's make an issue for it.